File: cat.js
#!/usr/bin/env node

const fs = require('fs');
const process = require("process")

const prefixArgs = process.argv[0].includes("node") ? 2 : 1;

if (process.argv.length > prefixArgs) {
    fs.readFile(process.argv[prefixArgs], "binary", function (err, text) {
        if (err) {
            return console.log("File not found");
        }
        console.log(text);
    });
} else {
    console.error("File not found")
    process.exit(1)
}

File: cat.go
package main

import (
	"fmt"
	"os"
	"io/ioutil"
	"log"
)

func main() {
	if (len(os.Args) > 1) {
		text, err := ioutil.ReadFile(os.Args[1])
		if (err != nil) {
			log.Fatal("File not found")
		}
		fmt.Println(string(text));
	} else {
		log.Fatal("File not found")
	}
}
